File: C/2020/56限/file2.c
//  file2.c
//  cl  file2.c
#include <stdio.h>
#define BUFSIZE 512

int main() {
    //
    FILE *fp;
    fp = fopen("file.c", "r");
    if (NULL == fp) {
        return 0;
    }

    //char *r;
    char buf[BUFSIZE];
    //r = fgets(buf, BUFSIZE, fp);
    //printf("%p, %p : %s\n", r, buf, buf);
    //while((r = fgets(buf, BUFSIZE, fp)) != NULL) {
    while(fgets(buf, BUFSIZE, fp) != NULL) {
        printf(": %s", buf);
    }

    //
    fclose(fp);

    return 0;
}

File: PhpBeginner/2021/EarlyPeriod(Sat.)/file_2.php
<?php   // file_2.php
//
//$fp = fopen('dummy', 'r');
$fp = fopen(__FILE__, 'r');
var_dump($fp);
//
while(false === feof($fp)) {
    $line = fgets($fp);
    var_dump($line);
}
/*
while(true) {
    $line = fgets($fp);
    if (false === $line) {
        break;
    }
    var_dump($line);
}
*/
/*
while(false !== ($line = fgets($fp))) {
    var_dump($line);
}
*/

//
fclose($fp);

